pkg/kubectl/resource/builder.go

```diff
@@ -587,7 +587,7 @@ func (b *Builder) visitorResult() *Result {
 		return &Result{singular: singular, visitor: visitors, sources: b.paths}
 	}
 
-	return &Result{err: fmt.Errorf("you must provide one or more resources by argument or filename")}
+	return &Result{err: fmt.Errorf(`you must provide one or more resources by argument or filename, try "kubectl help [command]" for help`)}
```

Comment:
Resource builder is not kubectl specific - this error needs to be handled above it.  Other commands will use resource.Builder which are not kubectl.